src/components/inputs/TextInput.js

```diff
@@ -420,48 +464,39 @@ export default class TextInput extends BaseInput {
     );
   }
 
+  getTopPaddings() {
+    const {floatingPlaceholder} = this.props;
+    return floatingPlaceholder ? (this.shouldShowTopError() ? undefined : 25) : undefined;
```

Comment:
let's avoid nested ternary condition. 
it's not readable 
also since we return undefined twice there's no need for it, the function by default will return undefined if there's no return value